azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/features/NetworkSecurityRuleApi.java

```diff
@@ -0,0 +1,93 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.azurecompute.arm.features;
+
+import org.jclouds.Fallbacks.EmptyListOnNotFoundOr404;
+import org.jclouds.Fallbacks.NullOnNotFoundOr404;
+import org.jclouds.azurecompute.arm.domain.NetworkSecurityRule;
+import org.jclouds.azurecompute.arm.domain.NetworkSecurityRuleProperties;
+import org.jclouds.azurecompute.arm.functions.FalseOn204;
+import org.jclouds.oauth.v2.filters.OAuthFilter;
+
+import org.jclouds.rest.annotations.RequestFilters;
+import org.jclouds.rest.annotations.QueryParams;
+import org.jclouds.rest.annotations.SelectJson;
+import org.jclouds.rest.annotations.Fallback;
+import org.jclouds.rest.annotations.PayloadParam;
+import org.jclouds.rest.annotations.ResponseParser;
+import org.jclouds.rest.annotations.MapBinder;
+import org.jclouds.rest.binders.BindToJsonPayload;
+
+import javax.inject.Named;
+import javax.ws.rs.Produces;
+import javax.ws.rs.Path;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.PUT;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.core.MediaType;
+import java.util.List;
+
+@Path("/resourcegroups/{resourcegroup}/providers/Microsoft.Network/networkSecurityGroups/{networksecuritygroup}")
+
+@QueryParams(keys = "api-version", values = "2016-03-30")
+@RequestFilters(OAuthFilter.class)
+@Consumes(MediaType.APPLICATION_JSON)
+public interface NetworkSecurityRuleApi {
+   @Named("networksecurityrule:create_or_update")
+   @Path("/securityRules/{networksecurityrulename}")
+   @PUT
+   @MapBinder(BindToJsonPayload.class)
+   @Fallback(NullOnNotFoundOr404.class)
+   @Produces(MediaType.APPLICATION_JSON)
+   NetworkSecurityRule create_or_update(@PathParam("networksecurityrulename") String ruleName,
+                                        @PayloadParam("properties") NetworkSecurityRuleProperties properties);
+
+   @Named("networksecurityrule:get_default_rule")
+   @Path("/defaultSecurityRules/{networksecurityrulename}")
+   @GET
+   @Fallback(NullOnNotFoundOr404.class)
+   NetworkSecurityRule get_default_rule(@PathParam("networksecurityrulename") String ruleName);
+
+   @Named("networksecurityrule:get")
+   @Path("/securityRules/{networksecurityrulename}")
+   @GET
+   @Fallback(NullOnNotFoundOr404.class)
+   NetworkSecurityRule get(@PathParam("networksecurityrulename") String ruleName);
+
+   @Named("networksecurityrule:delete")
+   @Path("/securityRules/{networksecurityrulename}")
+   @DELETE
+   @ResponseParser(FalseOn204.class)
+   boolean delete(@PathParam("networksecurityrulename") String ruleName);
```

Comment:
Since this delete is async, consider using `URIParser`, similar to [this](https://github.com/jclouds/jclouds-labs/blob/master/azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/features/ResourceGroupApi.java#L89-L94)

Please also test the returned uri to ensure it is not null. Also add the fallback test case like [this](https://github.com/jclouds/jclouds-labs/blob/master/azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/ResourceGroupApiMockTest.java#L141-L148)